app/questionnaire/rules.py

```diff
@@ -36,6 +38,53 @@ def evaluate_rule(when, answer_value):
     return result
 
 
+def evaluate_date_rule(when, answer_store, group_instance, metadata, answer_value):
+    result = False
+
+    match_value = None
+    date_comparison = when['date_comparison']
+
+    if 'value' in date_comparison:
+        if date_comparison['value'] == 'now':
+            match_value = datetime.utcnow().strftime('%Y-%m-%d')
+        else:
+            match_value = date_comparison['value']
+    elif 'id' in date_comparison:
+        match_value = get_answer_store_value(date_comparison['id'], answer_store, group_instance)
+    elif 'meta' in date_comparison:
+        match_value = get_metadata_value(metadata, date_comparison['meta'])
+
+    match_value = convert_to_datetime(match_value)
+
+    if 'offset_by' in date_comparison and match_value:
+        offset = date_comparison['offset_by']
+        match_value = match_value + relativedelta(days=offset.get('days', 0),
+                                                  months=offset.get('months', 0),
+                                                  years=offset.get('years', 0))
+
+    answer_value = convert_to_datetime(answer_value)
+    condition = when['condition']
+
+    if condition == 'equals' and match_value == answer_value:
+        result = True
+    elif condition == 'not equals' and match_value != answer_value:
+        result = True
+    elif condition == 'greater than' and isinstance(answer_value, datetime) and answer_value > match_value:
```

Comment:
This was to deal with a None answer value. Now i'm just going to return early for either answer_value, match_value or condition being None. This will satisfy your other comment above.